api2-war/src/main/java/eu/europeana/api2/utils/TurtleRecordWriter.java

```diff
@@ -0,0 +1,263 @@
+package eu.europeana.api2.utils;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+import org.apache.jena.datatypes.RDFDatatype;
+import org.apache.jena.rdf.model.Literal;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.Property;
+import org.apache.jena.rdf.model.RDFNode;
+import org.apache.jena.rdf.model.ResIterator;
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.rdf.model.StmtIterator;
+import org.apache.jena.vocabulary.RDF;
+import org.apache.jena.vocabulary.XSD;
+import static org.apache.commons.lang3.StringUtils.*;
+
+public class TurtleRecordWriter
+{
+    public static final int KB          = 1024;
+    public static       int BUFFER_SIZE = 32 * KB;
+
+    private BufferedWriter     _ps ;
+    private Map<String,String> _decl    = new HashMap();
+
+    public TurtleRecordWriter(OutputStream out) throws IOException
+    {
+        _ps  = new BufferedWriter(new OutputStreamWriter(out, "UTF-8")
+                                , BUFFER_SIZE);

```

Comment:
have made the close method and called it in the finally block of write method.

Reply:
Ok, this will work if we only call write once. We could be more flexible and allow multiple calls to write, but let's consider that when there is a need.

Reply:
I have had another look at this, and I'm still not happy about the solution for 2 reasons:
1. In the write method the call to writeHeader() method can still throw an exception because that's outside the try-catch block, so when that happens the writer is not closed. 
2. Nowhere in Java does a write operation also close the stream. We should adhere to standards as much as possible so code is easy to understand and use by the next Europeana developer (and by yourself in 1 year time when you forgot how you implemented this)

IMHO the best solution is to remove the call to close in write(), make TurtleRecordWriter autoclosable (implement the autoclosable interface) and move the call to write() in ObjectController to the try() block (there is one just above where write is called).

Reply:
I did try to make it auto closeable, But somehow it stopped working because of that. I will implement auto closeable once again and see closely.